server/src/modules/rules/constants/rules.constants.ts

```diff
@@ -49,8 +53,6 @@ export class RuleType {
       RulePossibility.SMALLER,
       RulePossibility.EQUALS,
       RulePossibility.NOT_EQUALS,
-      RulePossibility.CONTAINS,
```

Comment:
I think I understand this removal but just to check... Looking at doRuleAction it seems if the second value is not an array, which is extremely likely as you cannot provide a custom list for number, then we'll end up getting an exception as we'll try to call includes on it?

Reply:
Haha, I'll double check this in the morning. This also made me realise that I forgot to add some tests for the `throw`s, and also to make sure there is actually a catch somewhere to stop things blowing up

Reply:
the CONTAINS flow only calls `.includes` on val1 though, right? val2 could either be an array or a number and it'd still be fine 🤔 

looking here: https://github.com/jorenn92/Maintainerr/blob/0237cd4b43e722aef84849aab49df8aba2fdef02/server/src/modules/rules/helpers/rule.comparator.service.ts#L441

Reply:
Yes what you've done is fine. What would have happened is if val1 was a number and contains was used, it would run 2.includes(val2). Number doesn't have an includes() method so it would've thrown, get caught and return null instead.

I tested whether we'd need a migration and it's fine. Any rules configured with the removed ones would still execute but likely be broken because of the above issue anyway. When the rule UI is loaded the dropdown won't have anything selected so they'd need to reselect which is fine.